lib/index.js

```diff
@@ -128,6 +134,54 @@ module.exports = function(opts) {
           return String(recFile.contents);
         }
       }
+      
+          
+    // Seriously: you need more commments in your codes -.-
+    function loopHandler(inst) {
+        var args = /[^)"\']*["\']([^"\']*)["\'](,\s*([\s\S]*())){0,1}\s*/.exec(inst.args);
+        
+        if (args) {
+
+					var arr = JSON.parse(args[3]);
+					
+					if(arr) {
+						
+						var includePath = path.resolve(filebase, args[1]);
+						// for checking if we are not including the current file again
+						if (currentFilename.toLowerCase() === includePath.toLowerCase()) {
+							throw new Error('recursion detected in file: ' + currentFilename);
+						}
+
+						var includeContent = fs.readFileSync(includePath, 'utf-8');
+
+						if (opts.indent) {
+							includeContent = setIndent(inst.before, inst.before.length, includeContent);
+						}
+
+						// apply filters on include content
+						if (typeof opts.filters === 'object') {
+							includeContent = applyFilters(includeContent, args.input);
+						}
+
+						var recFile = new gutil.File({
+							cwd: process.cwd(),
+							base: file.base,
+							path: includePath,
+							contents: new Buffer(includeContent)
+						});
+						
+						var contents = '';
+						
+						for(var i in arr) {
```

Comment:
Yes... I was lazy in this case.

It could be better:

```js
for(var i in arr) {
	if(arr.hasOwnProperty(i)) {
		var context = arr[i];
		recFile = include(recFile, includeContent, args[3] ? context : {});
		if(typeof context == 'object') {
			context['_key'] = i;
		}
		contents += String(recFile.contents);
	}
}
```

Works with array or object. Flexible is better:
https://jsfiddle.net/webarthur/60ayy87z/1/